File: client/src/app/response/ResponseJson.ts

/**
 * エラー JSON の要素
 *
 * 対応する Java のクラス: vn.vnext.sefuri.sf.json.common.ItemErrorJson
 */
export interface ItemErrorJson {
    /** エラーコード */
    code: string;
    /** エラーメッセージ */
    message: string;
}

/**
 * エラー JSON
 *
 * 対応する Java のクラス: vn.vnext.sefuri.sf.json.common.ErrorJson
 */
export interface ErrorJson {
    /** エラーコード */
    code: string;
    /** エラー詳細 */
    items: ItemErrorJson[];
}

/**
 * 応答本文 JSON
 *
 * 対応する Java のクラス: vn.vnext.sefuri.sf.json.common.DataJson<T>
 */
export interface DataJson<T> {
    /** 応答本文 */
    data: T;
    /** メッセージコード */
    messageCode: string;
}

/**
 * 共通 JSON レスポンス
 *
 * @interface ResponseJson
 * @template T 本文の型
 */
export interface ResponseJson<T> {
    /** 応答本文 (成功時のみ) */
    res?: DataJson<T>;
    /** エラー (エラー時のみ) */
    error?: ErrorJson;
}

File: app/vn/vnext/sefuri/sf/json/common/ResponseJson.java
package vn.vnext.sefuri.sf.json.common;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * 応答 JSON
 *
 * 対応する TypeScript ソース client/src/app/response/ResponseJson.ts
 * Created by Teddy on 18/09/2016.
 */
public class ResponseJson<T extends AbstractJson> {

    @JsonProperty("res")
    private DataJson<T> res;

    @JsonProperty("error")
    private ErrorJson error;

    public DataJson<T> getRes() {
        return res;
    }

    public void setRes(DataJson<T> res) {
        this.res = res;
    }

    public ErrorJson getError() {
        return error;
    }

    public void setError(ErrorJson error) {
        this.error = error;
    }
}
